### DISCUSSION TITLE
dlopen, dlclose, dlsym

### DISCUSSION THREAD

**Discussion by tpimh** [pspdev/pspdev#83] [Ideas]
Following the recent discussion in Discord, these functions are lacking on PSP, but having them would be pretty cool. 

I suppose it's possible to implement dynamic libraries as PRX modules to be loaded at runtime. What is actually needed for this? What functions of the SDK map to `dlopen()`, `dlclose()` and `dlsym()`? Is there any example code for something like this?

Some games had taken [this approach](https://www.retroreversing.com/psp-debug-symbols) (perhaps we can learn something from studying them):

> Some games implement some major parts of the game logic in prx files rather than in the EBOOT executables. The PRX format contains lots of debug information as its basically a library.

**Comment by sharkwouter** [comment]
If you extract a umd, you'll see that most of them have some libraries included as prx files. This should be possible.

**Reply by tpimh** [reply]
They mostly look like Sony libs (e.g. `libfont.prx`) that are reused in different games. I am mostly interested in building my own dynamic libraries, however reusing those would also be cool (downside is they can't be redistributed, you must own a game to extract them from).
